backend/wmg/data/query.py

```diff
@@ -77,9 +77,31 @@ class MarkerGeneQueryCriteria(BaseModel):
     cell_type_ontology_term_id: str  # required!
 
 
+class WmgCubeQueryParams:
+    def __init__(self, cube_query_valid_attrs, cube_query_valid_dims):
+        self.cube_query_valid_attrs = cube_query_valid_attrs
+        self.cube_query_valid_dims = cube_query_valid_dims
+
+    def get_indexed_dims_to_lookup_query_criteria(self, cube: Array, pluralize: bool = True) -> list[str]:
+        return [self._transform_cube_index_name(i.name, pluralize) for i in cube.schema.domain]
+
+    def get_attrs_for_cube_query(self, cube: Array) -> list[str]:
+        return [i.name for i in cube.schema if i.name in self.cube_query_valid_attrs]
+
+    def get_dims_for_cube_query(self, cube: Array) -> list[str]:
+        return [i.name for i in cube.schema.domain if i.name in self.cube_query_valid_dims]
+
+    def _transform_cube_index_name(self, index_name: str, pluralize: bool = True) -> str:
+        if index_name != "organism_ontology_term_id" and pluralize:
+            return index_name + "s"
+
+        return index_name
+
+
 class WmgQuery:
-    def __init__(self, snapshot: WmgSnapshot) -> None:
+    def __init__(self, snapshot: WmgSnapshot, cube_query_params: WmgCubeQueryParams) -> None:
```

Comment:
just for my own understanding, under what circumstances would we not want to set the `cube_query_params` to be
```
WmgCubeQueryParams(
        cube_query_valid_attrs=READER_WMG_CUBE_QUERY_VALID_ATTRIBUTES,
        cube_query_valid_dims=READER_WMG_CUBE_QUERY_VALID_DIMENSIONS,
    )
    ```
?

Reply:
See the `calculate_markers.py` file in this PR for an example of when `WmgCubeQueryParams` is instantiated with `WRITER_*` instead of `READER_*`. That is the wmg pipeline uses the same query interface to construct cubes it has just finished building in order to build the `marker_genes_cube`.

However, the writer (pipeline) uses the latest schema to query the cube, whereas the reader (wmg api) might at a schema version that is `<=` the latest schema and therefore might have different `attrs` and `dimensions` to parse through as encoded in the `READER_*` constants